### PR TITLE
QVAC-17215 fix: preserve Portuguese UTF-8 handling in Chatterbox

### PR DESCRIPTION
**Pull request by ogad-tether**
## Summary
- fix UTF-8 string extraction at the JS/native boundary so Portuguese diacritics are preserved in Chatterbox input
- add focused JS and C++ regression coverage around UTF-8 string decoding and Portuguese Chatterbox input handling
- remove unrelated SDK TTS schema changes so this PR stays scoped to the native fix

## Test plan
- [x] `bare test/unit/chatterbox.inference.test.js` in `packages/qvac-lib-infer-onnx-tts`
- [x] `npm run test:cpp:build` in `packages/qvac-lib-infer-onnx-tts`
- [x] targeted native regressions confirmed: `ChatterboxLanguageModeTest.*Portuguese*` and `PreprocessDispatchTest.passesPortugueseDiacriticsThrough`
- [ ] full native integration suite still has one pre-existing environment-related failure in `ChatterboxEngineIntegrationTest.synthesizeProducesAudioWhenModelsAvailable`

### CODE REVIEW

File: packages/sdk/test/unit/text-to-speech-schema.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  } catch (error) {
+    t.ok(String(error).includes("Invalid option"));
+  }
+
+  try {
+    ttsSupertonicRuntimeConfigSchema.parse({
+      ttsEngine: "supertonic",
+      language: "ru",
+    });
+    t.fail("Supertonic schema should reject Russian");
```
Comment (GustavoA1604) [inline packages/sdk/test/unit/text-to-speech-schema.test.ts]: What are these supertonic tests for in this PR?

Comment (GustavoA1604) [reply packages/sdk/test/unit/text-to-speech-schema.test.ts]: Also including tests changes in sdk/ folder, should not happen

File: packages/qvac-lib-inference-addon-cpp/tests/js_utils_test.cpp [hunk 2]
```diff
@@ -11,12 +11,31 @@ namespace qvac_lib_inference_addon_cpp::js_utils {
...
+TEST(JsUtilsTest, StringRoundTripPreservesPortugueseDiacritics) {
+    js_env_t env;
+    const std::string original = asUtf8String(
+        u8"Olá mundo! Essa é uma demonstração de síntese de texto para voz usando Chatterbox");
+
+    auto jsString = js::String::create(&env, original);
+    const auto actual = js::String::fromValue(jsString).as<std::string>(&env);
+
+    EXPECT_EQ(actual, original);
+}
```
Comment (GustavoA1604) [inline packages/qvac-lib-inference-addon-cpp/tests/js_utils_test.cpp]: Not sure what is this test for?

**Review by GustavoA1604 — CHANGES_REQUESTED** [review]

File: packages/sdk/test/unit/text-to-speech-schema.test.ts (hunk 1 above)
Comment (ogad-tether) [reply packages/sdk/test/unit/text-to-speech-schema.test.ts]: Removed the SDK/schema changes and deleted the SDK-only test in c33811be so this PR stays scoped to the native UTF-8 fix. The PR body is updated to match the narrowed scope.

File: packages/qvac-lib-inference-addon-cpp/tests/js_utils_test.cpp (hunk 2 above)
Comment (ogad-tether) [reply packages/qvac-lib-inference-addon-cpp/tests/js_utils_test.cpp]: Kept this coverage but clarified it in c33811be. The bug is in the shared `String::as<std::string>()` UTF-8 extraction path, so this regression test exists to prove that path preserves the full UTF-8 byte sequence for Portuguese input.

**Comment by ogad-tether** [comment]
Addressed the requested changes on the latest branch revision:
- removed the unrelated SDK/schema changes and deleted the SDK-only test
- kept the shared C++ regression test but clarified that it covers the shared UTF-8 `String::as<std::string>()` extraction path
- updated the PR title/body to match the narrowed scope
- updated the PR branch with the latest `main`

Fresh checks are running on the updated branch.

**Review by GustavoA1604 — CHANGES_REQUESTED** [review]
This PR is touching TTS, onnx and  addon-cpp packages. Also the changes in TTS are all in tests, none in source code. Could you clarify if this is intended?

If so, we need to split the changes of each package in separate PRs and update qvac-registry-vcpkg for the changes in addon-cpp

**Comment by ogad-tether** [comment]
Scoped the PR down to the TTS addon package only. The actual UTF-8 fix now lives in \, and I removed the \ / \ package changes from this branch to avoid needing a registry update. The remaining changes are the TTS-package tests that verify Portuguese text is preserved and passed through correctly in the addon path.

**Comment by ogad-tether** [comment]
Follow-up clarification: this branch is now scoped to `packages/qvac-lib-infer-onnx-tts` only. The UTF-8 fix lives in `packages/qvac-lib-infer-onnx-tts/addon/src/addon/AddonJs.hpp`, and the PR no longer carries the `qvac-lib-infer-onnx` or `qvac-lib-inference-addon-cpp` package changes that would have required a registry update. The remaining changes are TTS-package tests that validate the Portuguese text path in this addon package.

**Comment by ogad-tether** [comment]
Addressed the scope concern on the latest branch revision.

What changed:
- the actual UTF-8 fix now lives in `packages/qvac-lib-infer-onnx-tts/addon/src/addon/AddonJs.hpp`
- the `qvac-lib-infer-onnx` and `qvac-lib-inference-addon-cpp` changes were removed from this PR
- the remaining test changes are all in `qvac-lib-infer-onnx-tts` and exist to verify the TTS addon path only

Why this addresses the review:
- the PR now touches only the TTS addon package
- there is now a real source change in the TTS package, not just tests
- no `qvac-registry-vcpkg` update is needed for this branch anymore because the fix no longer depends on a changed registry package

The remaining tests in the PR cover:
- Portuguese text reaches the addon boundary unchanged
- Portuguese text preprocessing stays lossless for TTS
- multilingual `[pt]` tokenization behavior remains correct in the TTS package

**Comment by ogad-tether** [comment]
@GustavoA1604 re-requesting your review on the latest revision. The PR is now scoped to the TTS addon package only, with the source fix living in \ and only TTS-package tests remaining.

**Comment by ogad-tether** [comment]
Correcting my previous note: @GustavoA1604 re-requesting your review on the latest revision. The PR is now scoped to the TTS addon package only, with the source fix in packages/qvac-lib-infer-onnx-tts/addon/src/addon/AddonJs.hpp and only TTS-package tests remaining.

**Review by GustavoA1604 — CHANGES_REQUESTED** [review]
Update CHANGELOG.md with [Unreleased] tag

**Comment by ogad-tether** [comment]
Addressed the latest review request on the branch head by updating `packages/qvac-lib-infer-onnx-tts/CHANGELOG.md` with an `[Unreleased]` section that includes this Portuguese UTF-8 fix.

**Review by GustavoA1604 — APPROVED** [review]

File: packages/qvac-lib-infer-onnx-tts/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+<<<<<<< HEAD
```
Comment (GustavoA1604) [inline packages/qvac-lib-infer-onnx-tts/CHANGELOG.md]: Wrong added lines

**Review by GustavoA1604 — CHANGES_REQUESTED** [review]

**Review by Zbig9000 — DISMISSED** [review]

**Review by GustavoA1604 — DISMISSED** [review]

**Comment by ogad-tether** [comment]
Reviewed the two failing native checks. The issues were: (1) `cpp-lint` failing on C++ formatting, and (2) `linux-x64-cpp-tests` failing because `ChatterboxLanguageModeTest.cpp` was calling `shouldUseEnglishMode(...)` with the newer two-argument signature while the branch header still had the older one-argument form. I updated the header/test-side formatting accordingly on the branch head.

**Review by freddy311082 — DISMISSED** [review]

**Comment by ogad-tether** [comment]
Addressed the remaining native build/prebuild failure. The shared root cause was two stale one-argument calls to `shouldUseEnglishMode(...)` in `addon/src/model-interface/ChatterboxEngine.cpp`; the branch now passes `language_` into those checks so the header/implementation/test signature is consistent again.

**Review by GustavoA1604 — APPROVED** [review]

**Review by Zbig9000 — APPROVED** [review]

**Review by ishanvohra2 — APPROVED** [review]
